verifier/src/x509.rs

```diff
@@ -64,6 +65,21 @@ impl TryFrom<String> for TrustAnchor {
     }
 }
 
+/// Try to get a trust anchor from a DER-encoded byte slice.
+///
+/// # Errors
+/// `Error::MbedTls` if the bytes are not a valid DER certificate.
+impl TryFrom<&[u8]> for TrustAnchor {
+    type Error = Error;
+
+    fn try_from(der: &[u8]) -> Result<Self> {
+        let mut certs = MbedtlsList::<Certificate>::new();
```

Comment:
I think its pretty confusing that DER is implied when the input is [u8] and PEM is implied when the input is a string since I can see a world in which someone passes PEM bytes by mistake. So my preference is to go with `try_from_der` and `try_from_pem` but I don't feel strongly about it.

Reply:
@eranrund bumping you for re-review since I did this refactor and the change is a bit more than a minor tweak